File: cdc-event-streaming/mastodon-consumer/src/main/java/com/onlyteo/sandbox/model/StatusEntity.java
package com.onlyteo.sandbox.model;

import jakarta.persistence.*;

import java.time.ZonedDateTime;

@Table(name = "STATUS")
@Entity
public class StatusEntity {

    @Id
    @Column(name = "id", nullable = false, unique = true)
    private String id;
    @Column(name = "url", nullable = false)
    private String url;
    @Column(name = "language", nullable = false)
    private String language;
    @Column(name = "visibility", nullable = false)
    private String visibility;
    @Column(name = "content", nullable = false)
    private String content;
    @Column(name = "in_reply_to_id", nullable = true)
    private String inReplyToStatusId;
    @Column(name = "in_reply_to_account_id", nullable = true)
    private String inReplyToAccountId;
    @Column(name = "replies_count", nullable = false)
    private Integer repliesCount;
    @Column(name = "reblogs_count", nullable = false)
    private Integer reblogsCount;
    @Column(name = "favourites_count", nullable = false)
    private Integer favouritesCount;
    @Column(name = "created_at", nullable = false)
    private ZonedDateTime createdAt;
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "account_id", referencedColumnName = "id")
    private AccountEntity account;

    public StatusEntity() {
    }

    public StatusEntity(String id,
                        String url,
                        String language,
                        String visibility,
                        String content,
                        String inReplyToStatusId,
                        String inReplyToAccountId,
                        Integer repliesCount,
                        Integer reblogsCount,
                        Integer favouritesCount,
                        ZonedDateTime createdAt) {
        this.id = id;
        this.url = url;
        this.language = language;
        this.visibility = visibility;
        this.content = content;
        this.inReplyToStatusId = inReplyToStatusId;
        this.inReplyToAccountId = inReplyToAccountId;
        this.repliesCount = repliesCount;
        this.reblogsCount = reblogsCount;
        this.favouritesCount = favouritesCount;
        this.createdAt = createdAt;
    }

    public String getId() {
        return id;
    }

    public String getUrl() {
        return url;
    }

    public String getLanguage() {
        return language;
    }

    public String getVisibility() {
        return visibility;
    }

    public String getContent() {
        return content;
    }

    public String getInReplyToStatusId() {
        return inReplyToStatusId;
    }

    public String getInReplyToAccountId() {
        return inReplyToAccountId;
    }

    public Integer getRepliesCount() {
        return repliesCount;
    }

    public Integer getReblogsCount() {
        return reblogsCount;
    }

    public Integer getFavouritesCount() {
        return favouritesCount;
    }

    public ZonedDateTime getCreatedAt() {
        return createdAt;
    }

    public AccountEntity getAccount() {
        return account;
    }

    public void setAccount(AccountEntity account) {
        this.account = account;
    }
}

File: spring-boot-cdc-event-streaming/mastodon-consumer/src/main/kotlin/com/onlyteo/sandbox/model/StatusEntity.kt
package com.onlyteo.sandbox.model

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table
import java.time.ZonedDateTime

@Table(name = "STATUS")
@Entity
class StatusEntity(
    @Id @Column(name = "id", nullable = false, unique = true)
    var id: String? = null,
    @Column(name = "url", nullable = false)
    val url: String? = null,
    @Column(name = "language", nullable = false)
    var language: String? = null,
    @Column(name = "visibility", nullable = false)
    var visibility: String? = null,
    @Column(name = "content", nullable = false)
    var content: String? = null,
    @Column(name = "in_reply_to_id", nullable = true)
    var inReplyToStatusId: String? = null,
    @Column(name = "in_reply_to_account_id", nullable = true)
    var inReplyToAccountId: String? = null,
    @Column(name = "replies_count", nullable = false)
    var repliesCount: Int? = null,
    @Column(name = "reblogs_count", nullable = false)
    var reblogsCount: Int? = null,
    @Column(name = "favourites_count", nullable = false)
    var favouritesCount: Int? = null,
    @Column(name = "created_at", nullable = false)
    var createdAt: ZonedDateTime? = null,
    @OneToOne(cascade = [CascadeType.ALL])
    @JoinColumn(name = "account_id", referencedColumnName = "id")
    var account: AccountEntity? = null
)